src/fetch-map/source.ts

```diff
@@ -0,0 +1,223 @@
+  REDUCED_QUERIES_TILE_RESOLUTION,
+} from '../constants-internal.js';
+import {
+  h3QuerySource,
+  H3QuerySourceOptions,
+  h3TableSource,
+  H3TableSourceOptions,
+  quadbinQuerySource,
+  QuadbinQuerySourceOptions,
+  quadbinTableSource,
+  QuadbinTableSourceOptions,
+  rasterSource,
+  vectorQuerySource,
+  VectorQuerySourceOptions,
+  vectorTableSource,
+  VectorTableSourceOptions,
+  vectorTilesetSource,
+  VectorTilesetSourceOptions,
+} from '../sources/index.js';
+import {Filter} from '../types.js';
+
+type FetchDatasetOptions = {
+  accessToken: string;
+  apiBaseUrl: string;
+  connection: string;
+  headers?: Record<string, string>;
+  localCache?: {
+    cacheControl: 'no-cache'[];
+  };
+  maxLengthURL?: number;
+};
+
+type FetchDataset = {
+  dataset: Dataset;
+  filters?: Filter;
+  options: FetchDatasetOptions;
+};
+
+// Copy of getCartoSource from cloud-native:
+// https://github.com/CartoDB/cloud-native/blob/main/workspace-www/src/features/common/utils/cartoDeckGL.ts#L79
+export function configureSource({
+  dataset,
+  filters,
+  options,
+}: FetchDataset): Promise<TilejsonResult> {
+  const {
+    geoColumn,
+    columns,
+    type,
+    source,
+    queryParameters,
+    aggregationExp,
+    aggregationResLevel: originalAggregationResLevel,
+  } = dataset;
+  const sourceOptions = getSourceOptions(options);
+  const spatialDataColumn = getColumnNameFromGeoColumn(geoColumn) || undefined;
+  const spatialIndex = geoColumn
+    ? getSpatialIndexFromGeoColumn(geoColumn)
+    : undefined;
+  const tileResolution = getDynamicTileResolution(spatialIndex);
+  const isH3 = spatialIndex === SpatialIndex.H3;
+  const isQuadbin = spatialIndex === SpatialIndex.QUADBIN;
+  let aggregationResLevel = originalAggregationResLevel;
+
+  if (typeof originalAggregationResLevel !== 'number' && isH3) {
+    aggregationResLevel = DEFAULT_AGGREGATION_RES_LEVEL_H3;
+  } else if (typeof originalAggregationResLevel !== 'number' && isQuadbin) {
+    aggregationResLevel = DEFAULT_AGGREGATION_RES_LEVEL_QUADBIN;
+  }
+
+  const spatialIndexOptions = {
+    aggregationExp: !aggregationExp ? DEFAULT_AGGREGATION_EXP : aggregationExp,
+    aggregationResLevel: scaleAggregationResLevel(
+      aggregationResLevel,
+      tileResolution
+    ),
+    spatialDataColumn,
+    ...(filters && {filters}),
+  } as H3QuerySourceOptions;
+  const tilesetOptions = {
+    ...sourceOptions,
+    tableName: source,
+  } as VectorTilesetSourceOptions;
+  const tableOptions = {
+    ...sourceOptions,
+    tableName: source,
+    tileResolution,
+  } as TableSourceOptions;
+  const queryOptions = {
+    ...sourceOptions,
+    sqlQuery: source,
+    tileResolution,
+    ...(queryParameters && {queryParameters}),
+  } as QuerySourceOptions;
+  const vectorOptions = {
+    spatialDataColumn,
+    ...(columns && {columns}),
+    ...(filters && {filters}),
+    ...(aggregationExp && {aggregationExp}),
+  } as VectorTableSourceOptions;
+
+  if (type === 'raster') {
```

Comment:
`SourceType` enum would be helpful

(yes, old `MapType`, which is actually type of source,, that's why i propose new name)

Reply:
Once I pull in the better type, this will use `MapType`: https://github.com/CartoDB/cloud-native/blob/main/workspace-www/src/features/maps/types.ts#L252

Is that what you want?